scripts/find-links-localize.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+#
+# Lists the `en` intra-site links on a localized Markdown page.
+# Checks if a localized page exists. Reports results.
+# Next step: modify the links on the page by prefixing the link with the <lang> and add/commit changes.
+# If there is no localized version of the page, validate the `en` link. Report results.
+# A temporary results file is created which lists the localized pages and the validated links.
+# To generate the file, run the script from the website/script directory with a <lang> parameter.
+#
+# Usage: python3 find-links-localize.py zh
+
+import argparse
+import glob
+import os
+import platform
+import re
+import sys
+import tempfile
+from pathlib import Path
+
+DIRS = ["concepts", "contribute", "home", "reference", "setup", "tasks", "tutorials"]
+TEST_DIRS = ["concepts"]
+tmp_file_output = {}
+
+def write_to_tmp():
+    """Write validation results to a temporary file."""
+    #count = 0
+    try:
+        fd, path = tempfile.mkstemp(
+            dir='/tmp' if platform.system() == 'Darwin' else tempfile.gettempdir(),
+            text=True
+        )
+        with open(fd, 'w') as f:
+            for path, path_output in tmp_file_output.items():
+                f.write("\n**File:**\n" + path + "\n")
+                for p in path_output:
+                    f.write(p)
+                    #count = count + 1
+    except OSError as ose:
+        print("[Error] Unable to create temp results file {}; error: {}"
+              .format(result_file, ose))
+        return -2
+    except IOError as e:
+        print(e)
+
+    #print(count)
+
+def find_site_links(content_dir, lang, sub_path):
+    """Find and validate intra-site links on a content page.
+       Look for all en links in localized page.
+       Links such as [...](/docs/...)
+       Check if a localized version exists
+       Replace the en version with the localized file
+       Report en links that are invalid
+       Investigating: Report localized links that are invalid
+    """
+
+    page_and_links = []
+
+    try:
+        with open(sub_path, "r") as mdFile:
+            content = mdFile.read()
+    except Exception as ex:
+        print("[Error] failed in reading markdown file: ".format(ex))
+
+    # Single results: searches for pattern: ](docs paths)
+    # remote_regex = re.compile(r"\]\(\/docs[\/\w\-\#\/]+\)")
+
+    # Single results: searches for pattern: []()
+    remote_regex = re.compile(r"(\[[\w\s\.-]*\])(\(\/docs[\/\w\-\#\/]+\))")
+    matches = remote_regex.findall(content)
+
+    for m in matches:
+        page_and_links.append("\nMatch link:" + m[0] + m[1])
+        # clean up path and split
+        subpaths = m[1][1:-1].split("/")
```

Comment:
Probably sample of output in this part will help why we need to split it this way.